### PR TITLE
Crash on client errors

### PR DESCRIPTION
**Pull request by scohen**
Our thrift clients accumulate processes when there is an error on the
server, This PR should help alleviate that by closing the clients if
they encounter an unexpected error.

@jparise 
@stemmler

### CODE REVIEW

File: lib/riffed/client.ex [hunk 1]
```diff
@@ -207,12 +223,17 @@ defmodule Riffed.Client do
           {:error, :closed} ->
             new_client = Client.reconnect(client)
             call_thrift(new_client, call_name, args, retry_count + 1)
-          err = {:error, _} ->
-            {new_client, err}
+
+          err = {:error, reason} ->
+            # try to close this, this process is going to die anyway.
```
Comment (jparise) [inline lib/riffed/client.ex]: Is this true of all non-`:closed` error reasons?

Comment (scohen) [inline lib/riffed/client.ex]: Yes, this is the change. Before we would handle them in a non-crashy way, but now we crash on all non-closed errors. 

I can take you through why we're going to do this, and I need to prove it out in production.

File: lib/riffed/client.ex [hunk 2]
```diff
@@ -183,11 +186,24 @@ defmodule Riffed.Client do
         {:reply, :ok, Client.new(client.connect)}
       end
 
+      def handle_info({:EXIT, _from, reason}, client=%Client{client: thrift_client}) do
```
Comment (jparise) [inline lib/riffed/client.ex]: `_reason` ?

File: lib/riffed/client.ex (hunk 1 above)
Comment (jparise) [inline lib/riffed/client.ex]: I follow the reasoning. The one thing I'm still fuzzy on is which cases close `thrift_client` here (on the next line) and which cases close the client through the trapped exit path.

**Comment by scohen** [comment]
@jparise, The underlying problem I'm trying to resolve is a bug in the thrift client that's fixed in 0.10 (whenever that comes out). 
This solves it at the wrong layer, though it's helpful if you're making a client for Elixir under 0.9.x. I'm not going to merge this unless people demand it.

**Pull request closed without merging**